compiler/ballerina-compiler-api/src/main/java/io/ballerina/compiler/api/symbols/Qualifiable.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.ballerina.compiler.api.symbols;
+
+import java.util.Set;
+
+/**
+ * An interface for symbols for which qualifiers apply.
+ *
+ * @since 2.0.0
+ */
+public interface Qualifiable {
+
+    /**
+     * Gets the set of qualifiers the user has applied to a symbol.
+     *
+     * @return The set of qualifiers applied to the symbol
+     */
+    Set<Qualifier> qualifiers();
```

Comment:
IIUC only the worker-symbol (may be not?) and xmlns-symbol doesn't have any qualifiers?
So can we move this method to the `Symbol` interface, and return an empty list for whoever doesn't have any qualifiers?
That would simplify the class hierarchy.. and can move this qualifiers in to the abstract `BallerinaSymbol` class as well in the implementation

Reply:
And module and service symbols as well though.

Reply:
If the qualifiers are not allowed, shouldn't we avoid exposing the API as well?

Reply:
Yeah, that was my thought as well. Hence this approach